test/unit_tests/chess_compiler_tests/01_precompiled_core_function/aie.mlir

```diff
@@ -22,12 +22,12 @@ module @test_chesss_01_precompiled_core_function {
   %lock13_3 = AIE.lock(%tile13, 3) { sym_name = "input_lock" }
   %lock13_5 = AIE.lock(%tile13, 5) { sym_name = "output_lock" }
 
-  func.func private @func(%A: memref<256xi32>, %B: memref<256xi32>) -> ()
+  func.func private @func(%A: memref<256xi32>, %B: memref<256xi32>, %lock: index) -> ()
```

Comment:
Perhaps `index` could be a more precise type depending on the architecture?
Or at least lowered to a more precise concrete type before calling the real AIE kernel?